tests/python/gaia-ui-tests/gaiatest/apps/gallery/regions/view_image.py

```diff
@@ -40,7 +40,8 @@ def tap_back_button(self):
         header.tap(x=20)
 
         # wait for the frame to close
-        Wait(self.marionette).until(lambda m: self.apps.displayed_app.name != 'Gallery')
+        from gaiatest.apps.Gallery.app import Gallery
+        Wait(self.marionette).until(lambda m: self.apps.displayed_app.manifest_url != Gallery.manifest_url)
```

Comment:
If this is the gallery app, then you can use self.manifest_url instead of Gallery.manifest_url.